### PR TITLE
Build and CI fixes

### PR DESCRIPTION
**Pull request by tiagolobocastro**
build: pin nixpkgs until we verify e2fs version requirement
    
    Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

---

    build: upgrade rustls crate
    
    This is to address a dependabot alert.
    
    Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

---

    test(bdd): always specify hostid
    
    Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

---

    test(python/ha): handle already connected error
    
    Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

---

    build: upgrade nvmeadm crate and nixpkgs
    
    Allows us to specify a hostid and a hostnqn prefix.
    Upgrades nixpkgs to get the latest nvme cli tool.
    Updates a few tests to use correct binary.
    
    Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

### CODE REVIEW

File: nix/pkgs/openapi-generator/default.nix [hunk 1]
```diff
@@ -28,7 +28,7 @@ let
       "find $out/.m2 -type f -regex '.+\\(\\.lastUpdated\\|resolver-status\\.properties\\|_remote\\.repositories\\)' -delete";
     outputHashAlgo = "sha256";
     outputHashMode = "recursive";
-    outputHash = if stdenv.hostPlatform.isDarwin then "sha256-8fk0lC3zBFgcGeBSASEDk0Zgz1xXLhJw6CSWszeSGf4=" else "sha256-mk6opj5FzELc/NambThC0K2ADJG2aRV4f8F2ZteMvMc=";
+    outputHash = if stdenv.hostPlatform.isDarwin then "" else "sha256-Z3jyn5WQpIXJz+8irsfxZ8LeQc2D+FZbAIZo7ME9OAw=";
```
Comment (tiagolobocastro) [inline nix/pkgs/openapi-generator/default.nix]: not to self: run on darwin to get the new hash

**Comment by tiagolobocastro** [comment]
bors try

**Comment by tiagolobocastro** [comment]
bors try

**Review by Abhinandan-Purkait — APPROVED** [review]

**Comment by tiagolobocastro** [comment]
bors try

**Comment by tiagolobocastro** [comment]
bors try

**Comment by tiagolobocastro** [comment]
bors merge

**Comment by tiagolobocastro** [comment]
bors cancel

File: nix/pkgs/openapi-generator/default.nix (hunk 1 above)
Comment (tiagolobocastro) [reply nix/pkgs/openapi-generator/default.nix]: Done!

**Comment by tiagolobocastro** [comment]
bors merge

**Comment by tiagolobocastro** [comment]
bors merge

**Pull request merged**